File: csharp/BCUR/BCUR.Tests/Crc32Tests.cs
using System.Text;

namespace BlockchainCommons.BCUR.Tests;

public class Crc32Tests
{
    [Fact]
    public void Crc32HelloWorld()
    {
        Assert.Equal(0xEBE6C6E6u, Crc32.Checksum(Encoding.UTF8.GetBytes("Hello, world!")));
    }

    [Fact]
    public void Crc32Wolf()
    {
        Assert.Equal(0x598C84DCu, Crc32.Checksum(Encoding.UTF8.GetBytes("Wolf")));
    }
}

File: swift/BCUR/Tests/BCURTests/Crc32Tests.swift
import Testing
@testable import BCUR

struct Crc32Tests {
    @Test func testCrc32() {
        #expect(Crc32.checksum(Array("Hello, world!".utf8)) == 0xEBE6_C6E6)
        #expect(Crc32.checksum(Array("Wolf".utf8)) == 0x598C_84DC)
    }
}
